remove `__version__`

davidism opened the pull request:
The `__version__` attribute is an old pattern from early in Python packaging. Setuptools eventually made it easier to use the pattern by allowing reading the value from the attribute at build time, and some other build backends have done the same.

However, there's no reason to expose this directly in code anymore. It's usually easier to use feature detection (`hasattr`, `try/except`) instead. `importlib.metadata.version("flask-debugtoolbar")` can be used to get the version at runtime in a standard way, if it's really needed.

This matches what we've already done in the Pallets projects.

jeffwidman commented on src/flask_debugtoolbar/__init__.py:
```diff
@@ -298,3 +297,19 @@ def teardown_request(self, exc: BaseException | None) -> None:
+    import warnings
+
+    if name == "__version__":
+        warnings.warn(
+            "The '__version__' attribute is deprecated and will be removed in"
+            " Flask-DebugToolbar 0.17. Use feature detection or"
+            " 'importlib.metadata.version(\"flask-debugtoolbar\")' instead.",
```
I'm personally also fine with just plain removing it in Flask DBT `0.16.0`... this is a dev tool, and low-volume usage, so we aren't going to break anyone's prod code.

If we just nuke it now, then it's one less thing to have to remember/spend time deleting later.

I'm 👍 either way.

davidism replied:
If you think it's ok to remove, I'll go for that, it definitely is more straightforward. This is currently zero ver after all, and it's probably unlikely people are using this extension's version for anything.

davidism commented:
I've removed `__version__` based on review. I also changed the Jinja extension compatibility to use feature detection instead of examining the version, which is exactly the advice I gave in the PR description ;-)
